mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/Game.kt

```diff
@@ -1,14 +1,63 @@
 package ch.epfl.sdp.mobile.application.chess
 
+import ch.epfl.sdp.mobile.application.chess.Color.Black
+import ch.epfl.sdp.mobile.application.chess.Color.White
+import ch.epfl.sdp.mobile.application.chess.Rank.*
+import ch.epfl.sdp.mobile.application.chess.implementation.PersistentGame
+import ch.epfl.sdp.mobile.application.chess.implementation.PersistentPieceIdentifier
+import ch.epfl.sdp.mobile.application.chess.implementation.buildBoard
+import ch.epfl.sdp.mobile.application.chess.rules.Action
+
 /**
  * An interface representing the current [Game], which contains a [Board] of pieces and on which
  * some [NextStep] should be performed to move forward for the chess game.
  */
 interface Game {
 
   /** The current [Board], which contains some pieces and should be rendered to the user. */
-  val board: Board
+  val board: Board<Piece<Color>>
 
   /** Represents the [NextStep] that must be performed on this [Game]. */
   val nextStep: NextStep
+
+  /**
+   * Returns a [Sequence] of the possible [Action] for the provided [Position]. If the [Position]
+   * does not have a [Piece] or it's not this player's turn, the resulting sequence might be empty.
+   *
+   * @param position the position for which the available [Action]s are queried.
+   * @return the [Sequence] of available actions.
+   */
+  fun actions(position: Position): Sequence<Action>
+}
+
+/** Creates a new [Game], with the standard starting positions for both players. */
+fun emptyGame(): Game {
+  val board =
+      buildBoard<Piece<Color>> {
+        var id = PersistentPieceIdentifier(0)
+
+        /** Populates a [row] with all the pieces of a given [color]. */
+        fun populateSide(row: Int, color: Color) {
+          set(Position(0, row), Piece(color, Rook, id++))
+          set(Position(1, row), Piece(color, Knight, id++))
+          set(Position(2, row), Piece(color, Bishop, id++))
+          set(Position(3, row), Piece(color, Queen, id++))
+          set(Position(4, row), Piece(color, King, id++))
+          set(Position(5, row), Piece(color, Bishop, id++))
+          set(Position(6, row), Piece(color, Knight, id++))
+          set(Position(7, row), Piece(color, Rook, id++))
+        }
+
+        // Populate the pieces.
+        populateSide(0, Black)
+        populateSide(7, White)
+
+        // Populate the pawns.
+        repeat(Board.Size) { column ->
+          set(Position(column, 1), Piece(Black, Pawn, id++))
+          set(Position(column, 6), Piece(White, Pawn, id++))
```

Comment:
Should the `inc()` method not be called instead of `id++`?

Reply:
The `inc()` method is an `operator fun` which is called when the increment operator is applied to `id` (it's essentially the reason why I defined `inc` on `PersistentPieceIdentifier`). So here `id++` is translated by the compiler to `id = id.inc()`.